src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperCompletionService.cs

```diff
@@ -298,7 +296,7 @@ void UpdateCompletions(string tagName, TagHelperDescriptor possibleDescriptor)
                     var descriptors = _tagHelperFactsService.GetTagHelpersGivenTag(
                         completionContext.DocumentContext,
                         prefixedName,
-                        completionContext.ContainingTagName);
+                        completionContext.ContainingParentTagName);
```

Comment:
Is this the root of the fix?

Reply:
Correct. The root problem seemed to be that we were checking `ContainingTagName` to check the binding. But in a scenario like:
```razor
<tagger>
    <|
</tagger>
```
`ContainingTagName` is not `"tagger"`, it is `""` because our context is inside the incomplete element. Therefore we must look to the `ContainingParentTagName` to find `"tagger"`.